src/test/integration/integrationtests.py

```diff
@@ -0,0 +1,151 @@
+
+        if "openHours" not in location["attributes"]:
+            logger.debug("doesn't have hours " + str(location["attributes"]["name"]))
+            return False
+
+        for day, open_hour_list in location["attributes"]["openHours"].iteritems():
+            valid_day_index = int(day) in valid_day_range
+            invalid_time_slot_count = len([i for i in open_hour_list if not self.is_valid_open_slot(i)])
+            logger.debug("day in range: " + str(valid_day_index))
+            if invalid_time_slot_count > 1 or not valid_day_index:
+                invalid_days += 1
+
+        logger.debug("invalid days: " + str(invalid_days) + " check against: " + str(min_valid_days))
+        return (7 - invalid_days) >= min_valid_days
+
+    def is_valid_open_slot(self, open_slot):
+        return open_slot is not None and open_slot["start"] is not None and open_slot["end"] is not None
+
+    def testOpenHoursUTC(self):
+        test_slot = None
+        test_diners = [diner for diner in restaurants["data"] if self.has_valid_open_hours(diner)]
+        open_hours = test_diners[0]["attributes"]["openHours"] 
+
+        logger.debug("test utc with diner: " + test_diners[0]["attributes"]["name"])
+
+        for day, open_hour_list in open_hours.iteritems():
+            valid_slots = [i for i in open_hour_list if self.is_valid_open_slot(i)]
+            if len(valid_slots) > 0:
+                test_slot = valid_slots[0]
+                logger.debug("test day slot day: " + str(day) + ", slot: " + str(test_slot))
+                break
+
+        regex = "[0-9]{4}-[0-9]{2}-[0-9]{2}[T][0-9]{2}:[0-9]{2}:[0-9]{2}[Z]"
+        self.assertRegexpMatches(test_slot["start"], regex)
+        self.assertRegexpMatches(test_slot["end"], regex)
+
+    # Tests that different verbs return expected responses
+    def test_verbs(self):
+        query_params = {}
+
+        self.assertEqual(query_request(extensions_url, access_token, "get", query_params).status_code, 200)
+
+        # DW not returning allowed methods headers, Apigee returning bad gateway
+        if query_request(extensions_url, access_token, "post", query_params).status_code == 502:
+            self.skipTest('DW not returning allowed methods headers that Apigee is expecting')
+
+        self.assertEqual(query_request(extensions_url, access_token, "post", query_params).status_code, 405)
+        self.assertEqual(query_request(extensions_url, access_token, "put", query_params).status_code, 405)
+        self.assertEqual(query_request(extensions_url, access_token, "delete", query_params).status_code, 405)
+
+    # Tests that a request with auth header returns a 401
+    def test_unauth(self):
+        self.assertEqual(unauth_request(extensions_url), 401)
+
+    # Tests that API response time is less than a value
+    def test_response_time(self):
+        self.assertLess(response_time(extensions_url, access_token), 2)
+        self.assertLess(response_time(dining_url, access_token), 3)
```

Comment:
This test failed for me because dining took 16 seconds to respond via api-dev.
```
======================================================================
FAIL: test_response_time (__main__.dw_tests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./integrationtests.py", line 116, in test_response_time
    self.assertLess(response_time(dining_url, access_token), 3)
AssertionError: 16.585891 not less than 3

----------------------------------------------------------------------
```